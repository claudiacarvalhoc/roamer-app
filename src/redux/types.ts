export const PROJECT_LOAD_SUCCESS = 'PROJECT_LOAD_SUCCESS';
export const PROJECT_LOAD_FAILED = 'PROJECT_LOAD_FAILED';
