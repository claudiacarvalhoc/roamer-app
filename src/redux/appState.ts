import { DefaultRootState } from "react-redux";

export interface AppState extends DefaultRootState {
    expand: boolean;
    projects: ProjectProperties[];
    menuTexts: MenuTexts,
    projectTexts: ProjectTextsProperties,
}

export interface MenuTexts {
    projectText: string;
    expandText: string;
    collapseText: string;
}

export interface ProjectTextsProperties {
    doneText: string;
    baseWordsText: string;
    teamText: string;
    keysText: string;
    qaIssuesText: string;
}

export interface IconType {
    Icon: 'up' | 'down' | 'check' | 'user' | 'rate' | 'camera' | 'book';
}

export interface TranslationSectionsProperties {
    country: string;
    language: string;
    done: number;
    wordsToDo: number;
    unverified: number;
}

export interface ProjectProperties {
    id: number;
    name: string;
    done: number;
    baseWords: number;
    team: number;
    keys: number;
    issues: number;
    icons: string[];
    badges: string[];
    translationSections: TranslationSectionsProperties[];
}
