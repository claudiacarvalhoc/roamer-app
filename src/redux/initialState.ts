import { AppState, MenuTextsState, ProjectState, ProjectTextsState } from "./appState";

export const menuInitialState: MenuTextsState = {
    projectText: 'New project ⇧⌘P',
    expandText: 'Expand all',
    collapseText: 'Collapse all',
};

export const projectsInitialState: ProjectTextsState = {
    doneText: 'DONE',
    baseWordsText: 'BASE WORDS',
    teamText: 'TEAM',
    keysText: 'KEYS',
    qaIssuesText: 'QA ISSUES',
    wordsToDoText: 'WORDS TO DO',
    unverifiedText: 'UNVERIFIED',
    addLanguageText: 'Add language',
};

export const projectInitialState: ProjectState = {
    id: 1,
    name: 'Roamer App (Android SDK test)',
    done: 30,
    baseWords: 160246,
    team: 26,
    keys: 1834,
    issues: 47514,
    icons: ['up', 'down', 'check', 'user', 'rate', 'camera', 'book'],
    badges: [
        {
            name: 'Roamer',
            type: 'orange',
        },
        {
            name: 'iOS',
            type: 'green',
        }
    ],
    translationSections: [{
        id: 1,
        country: 'uk',
        language: 'English',
        done: 20,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 2,
        country: 'uk',
        language: 'English',
        done: 20,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 3,
        country: 'uk',
        language: 'English',
        done: 20,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 4,
        country: 'uk',
        language: 'English',
        done: 20,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 5,
        country: 'ru',
        language: 'Russian',
        done: 45,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 6,
        country: 'ru',
        language: 'Russian',
        done: 45,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 7,
        country: 'ru',
        language: 'Russian',
        done: 45,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 8,
        country: 'ru',
        language: 'Russian',
        done: 45,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 9,
        country: 'it',
        language: 'Italian',
        done: 83,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 10,
        country: 'it',
        language: 'Italian',
        done: 83,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 11,
        country: 'it',
        language: 'Italian',
        done: 83,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 12,
        country: 'it',
        language: 'Italian',
        done: 83,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 13,
        country: 'mk',
        language: 'Macedonian',
        done: 67,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 14,
        country: 'mk',
        language: 'Macedonian',
        done: 67,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 15,
        country: 'mk',
        language: 'Macedonian',
        done: 67,
        wordsToDo: 16246,
        unverified: 50000,
    }]
};

export const appInitialState: AppState = {
    expand: true,
    projects: [
        projectInitialState
    ],
    MenuTextsState: menuInitialState,
    projectTexts: projectsInitialState,
};

export const newProjectInitialState: ProjectState = {
    id: 0,
    name: 'Roamer App (Android SDK test)',
    done: 0,
    baseWords: 160246,
    team: 26,
    keys: 1834,
    issues: 47514,
    icons: ['up', 'down', 'check', 'user', 'rate', 'camera', 'book'],
    badges: [
        {
            name: 'Roamer',
            type: 'orange',
        },
        {
            name: 'iOS',
            type: 'green',
        }
    ],
    translationSections: [],
};