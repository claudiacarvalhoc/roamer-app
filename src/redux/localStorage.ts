import { AppState } from './appState';
const KEY = 'state';

export const loadState = (): AppState => {
    try {
        const serializedState = localStorage.getItem(KEY);
        return serializedState !== null ? JSON.parse(serializedState) : undefined;
    } catch (err) {
        console.error(`Error loading state from local storage`, err);
        return undefined;
    }
};

export const saveState = (state: AppState) => {
    try {
        const serializedState = JSON.stringify(state);
        localStorage.setItem(KEY, serializedState);
    } catch (err) {
        console.error(`Error saving state on local storage`, err);
    }
};