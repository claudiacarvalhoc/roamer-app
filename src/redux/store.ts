import { AnyAction } from 'redux';
import { AppState } from './appState';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { applyMiddleware, compose, createStore } from 'redux';
import { createLogger } from 'redux-logger';
import { rootReducer } from './reducers';
import thunk from 'redux-thunk';
import { loadState, saveState } from './localStorage';

export type AppThunk<R> = ThunkAction<R, AppState, unknown, AnyAction>;
export type AppDispatch = ThunkDispatch<AppState, unknown, AnyAction>;


export const configureStore = (initialState: AppState) => {
    const logger = createLogger();
    const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const middleware = [thunk, logger];

    const preloadedState = loadState() ?? initialState;
    const store = createStore(
        rootReducer,
        {
            app: preloadedState,
        },
        composeEnhancers(applyMiddleware(...middleware)),
    );

    store.subscribe(() => {
        saveState(store.getState().app);
    });

    return store;
};

export default configureStore;
