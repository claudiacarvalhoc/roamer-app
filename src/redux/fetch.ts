import { AppDispatch } from './store';
import { addProjectAction } from './app/actions';
import { newProjectInitialState } from './initialState';

export const createProject = (): ((dispatch: AppDispatch) => Promise<void>) => {
    return dispatch => {
        dispatch(addProjectAction(newProjectInitialState));
        return Promise.resolve();
    };
};