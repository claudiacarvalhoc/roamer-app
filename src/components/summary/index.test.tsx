import React from 'react';
import Summary from './index';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { appInitialState, newProjectInitialState } from '../../redux/initialState';
import { ProjectState } from '../../redux/appState';

const getShallowWrapper = (project: ProjectState) => {
    const store = configureMockStore()({
        app: {
          ...appInitialState,
          projects: [
            project
          ]
        }
      });
      return shallow(
          <Provider store={store}>
              <Summary project={project} />
          </Provider>
      );
};

describe('<Summary />', () => {
  it('render correctly', () => {
    const component = getShallowWrapper(newProjectInitialState);
    expect(toJson(component)).toMatchSnapshot();
  });
  it('when badges has elements, the badges are visible', () => {
    const project = {
      ... newProjectInitialState,
      badges: [],
    };
    const component = getShallowWrapper(project);
    expect(component.find('[data-test="summary-badges"]').exists()).toBeFalsy();
  });
  it('when badges is empty, the badges are not visible', () => {
    const project = {
      ... newProjectInitialState,
      badges: [],
    };
    const component = getShallowWrapper(project);
    expect(component.find('[data-test="summary-badges"]').exists()).toBeFalsy();
  });
});
