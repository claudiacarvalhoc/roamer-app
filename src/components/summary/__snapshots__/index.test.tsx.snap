// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Summary)
    project={
      Object {
        "badges": Array [
          Object {
            "name": "Roamer",
            "type": "orange",
          },
          Object {
            "name": "iOS",
            "type": "green",
          },
        ],
        "baseWords": 160246,
        "done": 0,
        "icons": Array [
          "up",
          "down",
          "check",
          "user",
          "rate",
          "camera",
          "book",
        ],
        "id": 0,
        "issues": 47514,
        "keys": 1834,
        "name": "Roamer App (Android SDK test)",
        "team": 26,
        "translationSections": Array [],
      }
    }
  />
</ContextProvider>
`;
